packages/runtime-dom/src/components/Transition.ts

```diff
@@ -123,6 +123,7 @@ export function resolveTransitionProps(
       const resolve = () => finishEnter(el, isAppear, done)
       hook && hook(el, resolve)
       nextFrame(() => {
+        enableTransition(el)
```

Comment:
This (and `disableTransition` calls in beforeEnter and beforeAppear) seems unnecessary. When elements get a class in a none-rendered state (`display: none` or off-DOM) they do not trigger transitions anyway.

Reply:
Oh ok I see you are doing this so it can support `visibility` toggles.

Reply:
Testing this locally, I find that this actually alters the behavior of `v-show` transitions when you hit toggle really fast.

The current behavior smoothly transitions half-way into the opposite direction, but with this change it hard resets to the enter from state.